TypeDBConsole.java

```diff
@@ -651,18 +656,43 @@ private void runQueryPrintAnswers(TypeDBTransaction tx, TypeQLQuery query) {
         if (query instanceof TypeQLDefine) {
             tx.query().define(query.asDefine()).get();
             printer.info("Concepts have been defined");
+            hasUncommittedChanges = true;
         } else if (query instanceof TypeQLUndefine) {
             tx.query().undefine(query.asUndefine()).get();
             printer.info("Concepts have been undefined");
+            hasUncommittedChanges = true;
         } else if (query instanceof TypeQLInsert) {
             Stream<ConceptMap> result = tx.query().insert(query.asInsert());
-            printCancellableResult(result, x -> printer.conceptMap(x, tx));
+            AtomicBoolean changed = new AtomicBoolean(false);
+            printCancellableResult(result, x -> {
+                changed.set(true);
+                printer.conceptMap(x, tx);
+            });
+            if (changed.get()) hasUncommittedChanges = true;
         } else if (query instanceof TypeQLDelete) {
-            tx.query().delete(query.asDelete()).get();
-            printer.info("Concepts have been deleted");
+            Stream<ConceptMap> result = tx.query().match(query.asDelete().match());
+            AtomicInteger answerCount = new AtomicInteger();
+            printCancellableResult(result, x -> {
+                answerCount.getAndIncrement();
+                printer.conceptMap(x, tx);
+            });
+            if (answerCount.get() > 0) {
+                tx.query().delete(query.asDelete()).get();
+                printer.info("Concepts have been deleted");
+                hasUncommittedChanges = true;
+            } else {
+                printer.info("No concepts were matched");
+            }
         } else if (query instanceof TypeQLUpdate) {
+            Stream<ConceptMap> matchResult = tx.query().match(query.asUpdate().match());
+            AtomicInteger answerCount = new AtomicInteger();
+            printCancellableResult(matchResult, x -> {
+                answerCount.getAndIncrement();
+                printer.conceptMap(x, tx);
+            });
             Stream<ConceptMap> result = tx.query().update(query.asUpdate());
             printCancellableResult(result, x -> printer.conceptMap(x, tx));
+            if (answerCount.get() > 0) hasUncommittedChanges = true;
```

Comment:
aren't we going to print everything twice now?

Reply:
Ah yes. ![image](https://user-images.githubusercontent.com/14253102/139835650-79b02939-96b6-424c-a6a1-a8df3ef2a4e7.png)